### ISSUE TITLE
Backup and restore of add on with no data fails

### ISSUE THREAD

**Issue by adamurban** [home-assistant/supervisor#3252]
### Describe the issue you are experiencing

If I backup an add on like ESP Home that commonly has no data, the backup file created contains the wrong file structure, and restoration from this backup fails.

Eg:

```
21-10-22 03:02:11 ERROR (MainThread) [supervisor.jobs] Unhandled exception: [Errno 2] No such file or directory: '/data/tmp/tmpclnxs8ta/data'
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 106, in wrapper
    return await self._method(*args, **kwargs)
  File "/usr/src/supervisor/supervisor/addons/__init__.py", line 360, in restore
    await addon.restore(tar_file)
  File "/usr/src/supervisor/supervisor/addons/addon.py", line 868, in restore
    await self.sys_run_in_executor(_restore_data)
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/supervisor/supervisor/addons/addon.py", line 862, in _restore_data
    shutil.copytree(Path(temp, "data"), self.path_data, symlinks=True)
  File "/usr/local/lib/python3.9/shutil.py", line 563, in copytree
    with os.scandir(src) as itr:
FileNotFoundError: [Errno 2] No such file or directory: '/data/tmp/tmpclnxs8ta/data'
21-10-22 03:02:11 WARNING (MainThread) [supervisor.backups.backup] Can't restore Add-on a0d7b954_esphome: 
21-10-22 03:02:11 INFO (MainThread) [supervisor.backups.manager] Partial-Restore ae36c8ce done
```

If I look at a simple example backup tarball, it looks like this:
<img width="394" alt="Screen Shot 2021-10-22 at 3 01 57 AM" src="https://user-images.githubusercontent.com/3276234/138408753-e8543cb1-7669-4930-a74a-d757bdefec78.png">

```
% tar -tf October_22_2021.tar
./
./a0d7b954_esphome.tar.gz
./backup.json
```
And if I look at the inner .tar.gz, I see this:
```
% tar -tf October_22_2021/a0d7b954_esphome.tar.gz
./
./addon.json
```

Instead of the expected data directory/structure, which should look something like this:
```
% tar -tf October_22_2021/a0d7b954_nut.tar.gz 
./
./addon.json
data/
data/options.json

```

### What is the used version of the Supervisor?

supervisor-2021.10.0

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### What is the version of your installed operating system?

6.5

### What version of Home Assistant Core is installed?

core-2021.10.6

### Steps to reproduce the issue

1) Create a backup from an add on like ESP Home that isn't configured with any options
2) Try to restore that backup

### Anything in the Supervisor logs that might be useful for us?

```txt
21-10-22 03:02:11 ERROR (MainThread) [supervisor.jobs] Unhandled exception: [Errno 2] No such file or directory: '/data/tmp/tmpclnxs8ta/data'
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 106, in wrapper
    return await self._method(*args, **kwargs)
  File "/usr/src/supervisor/supervisor/addons/__init__.py", line 360, in restore
    await addon.restore(tar_file)
  File "/usr/src/supervisor/supervisor/addons/addon.py", line 868, in restore
    await self.sys_run_in_executor(_restore_data)
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/supervisor/supervisor/addons/addon.py", line 862, in _restore_data
    shutil.copytree(Path(temp, "data"), self.path_data, symlinks=True)
  File "/usr/local/lib/python3.9/shutil.py", line 563, in copytree
    with os.scandir(src) as itr:
FileNotFoundError: [Errno 2] No such file or directory: '/data/tmp/tmpclnxs8ta/data'
21-10-22 03:02:11 WARNING (MainThread) [supervisor.backups.backup] Can't restore Add-on a0d7b954_esphome: 
21-10-22 03:02:11 INFO (MainThread) [supervisor.backups.manager] Partial-Restore ae36c8ce done
```

**Comment by appleguru** [comment]
Hmm, I think related to this change I was unable to start esp home on the latest system... When I try too start the add on, I see:
```21-11-01 22:13:08 ERROR (MainThread) [supervisor.utils.json] Can't write /data/addons/data/a0d7b954_esphome/options.json: [Errno 2] No such file or directory: '/data/addons/data/a0d7b954_esphome/tmp14muovhc'
21-11-01 22:13:08 ERROR (MainThread) [supervisor.addons.addon] Add-on a0d7b954_esphome can't write options```
In the supervisor.

And indeed, if I log into the container, eg, `docker exec -it hassio_supervisor bash`, the `/data/addons/data/a0d7b954_esphome/` folder does not exist.

I did  `mkdir /data/addons/data/a0d7b954_esphome/` in the supervisor's container and was then able to start the add-on without issue.

**Comment by agners** [comment]
@appleguru we've seen that issue with ESPHome Add-on in the past. But I was hoping that this fix should address that as well. From the date it seems that happened recently, so you should already have the fix on your system (got fixed in 2021.10.5, [2021.10.6 got pushed to stable on 2021-10-25](https://github.com/home-assistant/version/commit/65008372d2766d0d6c54f4e1332d337fbe27fda1#diff-c28bc2dee7e0adeee098ecb5d0ee64d65c8128ef7cdcc792f7c401b0f614b1f2)).

**Comment by adamurban** [comment]
> @appleguru we've seen that issue with ESPHome Add-on in the past. But I was hoping that this fix should address that as well. From the date it seems that happened recently, so you should already have the fix on your system (got fixed in 2021.10.5, [2021.10.6 got pushed to stable on 2021-10-25](https://github.com/home-assistant/version/commit/65008372d2766d0d6c54f4e1332d337fbe27fda1#diff-c28bc2dee7e0adeee098ecb5d0ee64d65c8128ef7cdcc792f7c401b0f614b1f2)).

This happened while running 2021.10.6, as well as after upgrading to 2021.10.7 (including after making a backup), so I don't think the PR resolved it unfortunately. The "manual" fix does seem to work though, so hopefully it doesn't happen again...
